### DISCUSSION TITLE
Refs not behaving the same in dev and prod

### DISCUSSION THREAD

**Discussion by Zaytt** [vercel/next.js#60745] [Help] [answered]
### Summary
Hi!
I'm doing this Next.js [tutorial](https://youtu.be/Yw7yWHigGKI?si=rht5Rzx7kYwF7dCG&t=8758)
I managed to implement the desired functionality of the scrolling line and the closing circles.

It works as expected when running locally in dev, but an issue arises when running in prod.
The animated closing circles are all synchronized when running in prod, both locally and when deployed at Vercel.
It seems as if they were all using the same ref instead of their unique assigned ref.

Added a sandbox and example repo that illustrates the issue. Just have to compare scrolling down when running dev vs prod.

### Example

https://codesandbox.io/p/github/Zaytt/nextjs-framer-scroll-test
https://github.com/Zaytt/nextjs-framer-scroll-test

**Comment by icyJoseph** [comment, marked as answer]
Hi,

For some reason it is `useScroll` that **_seems_** to be buggy  (not 100% confirmed). I haven't looked too much into their implementation, or seeing if its still somehow Next.js, or React, or framer-motion specifically that has the issue:

- https://github.com/framer/motion/issues/2483

A couple of things you can try:

```js
const Details = ({ position, company, time, address, work }) => {
  const [ref, setRef] = useState(null);

  return (
    <li
      ref={setRef}
      className="my-8 first:mt-0 last:mb-8 w-[60%] mx-auto flex flex-col items-center justify-between"
    >
      <LiIcon reference={{ current: ref }} />

```

> Though I'd also strongly suggest using `forwardRef` on `LiIcon`

This would essentially trigger a re-render and update the value seen by `LiIcon` and through that, `useScroll`.

Or you could pass the scroll progress:

```js
const Details = ({ position, company, time, address, work }) => {
  const ref = useRef(null);

  const { scrollYProgress } = useScroll({
    target: ref,
    offset: ['center end', 'center center'],
  });

  return (
    <li
      ref={ref}
      className="my-8 first:mt-0 last:mb-8 w-[60%] mx-auto flex flex-col items-center justify-between"
    >
      <LiIcon scrollYProgress={scrollYProgress} />
```

And that should work too...

On Dev Mode it looks like it works, because of Strict Mode I believe, but if you disable ReactStrictMode is fails too. Kind of weird though I am gonna see if I can pin point what's happening, starting from `useScroll`.

**Reply by Zaytt** [reply]
I'll go for passing the scrollYProgress. That indeed did the trick.
Thanks for the help, Joseph!

**Reply by icyJoseph** [reply]
No worries, please don't forget to mark as answered if you are satisfied with the answer :)
